hermes/crates/cbork/cddl-parser/tests/byte_sequences.rs

```diff
@@ -1,91 +1,98 @@
-// cspell: words hexpair rstuvw abcdefghijklmnopqrstuvwyz rstuvw
+// cspell: words hexpair rstuvw abcdefghijklmnopqrstuvwyz rstuvw Xhhb Bhcm
 
 use cddl_parser::{
     self,
     cddl_test::{CDDLTestParser, Parser, Rule},
 };
 
-#[test]
-/// Test if the `HEX_PAIR` rule passes properly.
-fn check_hexpair() {
-    let hex_pairs = vec!["00", "ab", "de", "0f", "f0"];
+pub(crate) const HEXPAIR_PASSES: &[&str] = &["00", "ab", "de", "0f", "f0"];
+
+pub(crate) const HEXPAIR_FAILS: &[&str] = &["0", " 0", "0 ", "az", "0p"];
+
+pub(crate) const URL_BASE64_PASSES: &[&str] = &[
+    "abcdefghijklmnopq   rstuvw   yz01\t23456789-_ABCDEFGHIJKLMNOPQRSTUVWXYZ~",
+    "abcdefghijklmnopqrstuvwyz0123456789-_ABCDEFGHIJKLMNOPQRSTUVWXYZ",
+];
+
+pub(crate) const URL_BASE64_FAILS: &[&str] = &[
+    "abcdefghijklmnopq #  rstuvw   yz01\t23456789-_ABCDEFGHIJKLMNOPQRSTUVWXYZ~ ",
+    "abcdefghijklmnopq $  rstuvw   yz01\t23456789-_ABCDEFGHIJKLMNOPQRSTUVWXYZ~\t",
+    "abcdefghijklmnopq %  rstuvw   yz01\t23456789-_ABCDEFGHIJKLMNOPQRSTUVWXYZ~\n",
+    "abcdefghijklmnopq ^  rstuvw   yz01\t23456789-_ABCDEFGHIJKLMNOPQRSTUVWXYZ~\r",
+    "abcdefghijklmnopq &  rstuvw   yz01\t23456789-_ABCDEFGHIJKLMNOPQRSTUVWXYZ~\r\n",
+];
+
+pub(crate) const BYTES_PASSES: &[&str] = &[
+    "h''",
+    "b64''",
+    "''",
+    "h'00'",
+    "h'63666F6FF6'",
+    "h'68656c6c6f20776f726c64'",
+    "h'4 86 56c 6c6f'",
+    "h' 20776 f726c64'",
+    "h'00112233445566778899aabbccddeeff0123456789abcdef'",
+    "h'0 1 2 3 4 5 6 7 8 9 a b c d e f'",
+    "h' 0 1 2 3 4 5\r 6 7 \n 8 9 a\r\n\t b c d e f'",
+    "h'0 \n\n\r f'",
+    "b64'aHR0cHM6Ly93d3cuZXhhbXBsZS5jb20vcGFnZT9wYXJhbTE9dmFsdWUxJnBhcmFtMj12YWx1ZTI~'",
+    "'text\n that gets converted \\\' into a byte string...'",
+];
 
-    let not_hex_pairs = vec!["0", " 0", "0 ", "az", "0p"];
+pub(crate) const BYTES_FAILS: &[&str] = &[
+    "h64",
+    "b64",
+    "\"\"",
+    "h ''",
+    "b64 ''",
+    "h'001'",
+    "b64'abcdefghijklmnopq #  rstuvw   yz01\t23456789-_ABCDEFGHIJKLMNOPQRSTUVWXYZ~'",
+    "b64'abcdefghijklmnopq   & rstuvw   yz01\t23456789-_ABCDEFGHIJKLMNOPQRSTUVWXYZ'",
+    "'\u{7}'",
+];
 
-    for hp in hex_pairs {
-        let parse = CDDLTestParser::parse(Rule::HEX_PAIR, hp);
+/// # Panics
+pub(crate) fn passes_tests_rule(rule_type: Rule, test_data: &[&str]) {
+    for test in test_data {
+        let parse = CDDLTestParser::parse(rule_type, test);
         assert!(parse.is_ok());
     }
+}
 
-    for hp in not_hex_pairs {
-        let parse = CDDLTestParser::parse(Rule::HEX_PAIR, hp);
+/// # Panics
+pub(crate) fn fails_tests_rule(rule_type: Rule, test_data: &[&str]) {
+    for test in test_data {
+        let parse = CDDLTestParser::parse(rule_type, test);
         assert!(parse.is_err());
     }
 }
```

Comment:
Because we often test both a passes set and a fails set at the same time,  you can also make a `check_tests_rule(rule_type, passes, fails)` which checks the same rule against the passes and fails data sets.
It should just run these two tests that you created here.  
You probably also don't need to re-assign the passes/fails data sets unless you are composing a new set from multiple sets.

So you could just go:

```
check_tests_rule(Rule::HEX_PAIR, HEXPAIR_PASSES, HEXPAIR_FAILS);
```
etc.

Reply:
You should also move these to their own file, as they are universal test operations, not specific to byte_sequences.
See: https://doc.rust-lang.org/rust-by-example/testing/integration_testing.html
for recommended way to share utility functions with integration tests.